File: Semester 4/Programming and Development Methods/CSProject/Project/Project/Domain/RaceType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Project.Domain
{
    class RaceType : IdEntity<KeyValuePair<RaceLength, RaceStyle>>
    {
        public RaceType(RaceLength length, RaceStyle style)
        {
            this.length = length;
            this.style = style;
        }

        public RaceLength length { get; set; }
        public RaceStyle style { get; set; }



        public KeyValuePair<RaceLength, RaceStyle> GetId()
        {
            return new KeyValuePair<RaceLength, RaceStyle>(length, style);
        }

        public void SetId(KeyValuePair<RaceLength, RaceStyle> id)
        {
            this.length = id.Key;
            this.style = id.Value;
        }
    }
}

File: Semester 4/Programming and Development Methods/MPPProject/src/main/java/Domain/RaceType.java
package Domain;


public class RaceType implements IdEntity<Domain.Pair<RaceLength,Style>> {

    public RaceLength len;
    public Style style;

    public RaceType(RaceLength len, Style style) {
        this.len = len;
        this.style = style;
    }

    public RaceLength getLen() {
        return len;
    }

    public void setLen(RaceLength len) {
        this.len = len;
    }

    public Style getStyle() {
        return style;
    }

    public void setStyle(Style style) {
        this.style = style;
    }


    @Override
    public Pair<RaceLength, Style> getId() {
        return new RaceTypeLenPair(len,style);
    }

    @Override
    public void setId(Pair<RaceLength, Style> raceLengthStylePair) {
        this.len = raceLengthStylePair.getFirst();
        this.style = raceLengthStylePair.getSecond();
    }
}
